### PR TITLE
chore(deps): upgrade PM2

### PR DESCRIPTION
**Pull request by jamesgorrie**
## What does this change?

Runs `yarn add --force pm2` to reinstall `pm2@5.3.0` with [their dependency vulnerabilities fixed](https://github.com/Unitech/pm2/issues/5643).

`pm2` don't bump their package but rather fix in place.

The snyk reports were generated locally by running `snyk monitor dotcom-rendering` hence they're in the `guardian-people` project.

## Why?

To have no high vulnerabilities.

## Screenshots

| Before      | After      |
| ----------- | ---------- |
| ![before][] | ![after][] |
| [Snyk report][snyk-before] | [Snyk report][snyk-after] |

[before]: https://github.com/guardian/dotcom-rendering/assets/31692/5c1fd6ba-e58e-494b-bfd4-18dc28239282
[after]: https://github.com/guardian/dotcom-rendering/assets/31692/7ada1afb-1727-4198-8bdd-60f5b9c28ea2

[snyk-before]: https://app.snyk.io/org/guardian-people/project/1b1050f5-afed-47ed-b28e-1fc7d36f453f/history/46f75a78-e139-477c-8555-e999024246f9
[snyk-after]: https://app.snyk.io/org/guardian-people/project/1b1050f5-afed-47ed-b28e-1fc7d36f453f/history/e6f902a1-c25f-4e49-a48a-c971caf16a49

### CODE REVIEW

**Review by arelra — COMMENTED** [review]
Actually I think the pm2 version we use on our live instances is installed to [/usr/local/node/pm2](https://github.com/guardian/dotcom-rendering/blob/4f71c6877593b3179e1faa38c6e8fb1a793cccf9/dotcom-rendering/makefile#L58) as part of the AMI. Would that always be the latest (i.e fixed) version?

**Comment by jamesgorrie** [comment]
> Actually I think the pm2 version we use on our live instances is installed to [/usr/local/node/pm2](https://github.com/guardian/dotcom-rendering/blob/4f71c6877593b3179e1faa38c6e8fb1a793cccf9/dotcom-rendering/makefile#L58) as part of the AMI. Would that always be the latest (i.e fixed) version?

@arelra I think it definitely is, so technically we don't actually have this vulnerability on the live instances. 

Saying that, I do think we should remove that from those boxes as it's a very opaque dependency and could cause any number of hard-to-debug issues through being implicitly used there.

I think we would need to ensure we can the installed `pm2` on those boxes before doing that though.

**Comment by arelra** [comment]
> @arelra I think it definitely is, so technically we don't actually have this vulnerability on the live instances.

Great thanks for confirming.
 
> Saying that, I do think we should remove that from those boxes as it's a very opaque dependency and could cause any number of hard-to-debug issues through being implicitly used there.

Agreed we should be using the version that is in our dependencies.

> I am going to merge this as it remove our reported, if not real, vulnerabilities in Snyk.

Thanks for sorting this!

File: yarn.lock [hunk 1]
```diff
@@ -16185,7 +16185,12 @@ semver-truncate@^3.0.0:
   resolved "https://registry.yarnpkg.com/semver/-/semver-5.7.2.tgz#48d55db737c3287cd4835e17fa13feace1c41ef8"
   integrity sha512-cBznnQ9KjJqU67B52RMC65CMarK2600WFnbkcaiwWq3xy/5haFJlshgnpjovMVJ+Hff49d8GEn0b87C5pDQ10g==
 
-semver@6.3.0, semver@6.3.1, semver@^6.0.0, semver@^6.1.1, semver@^6.1.2, semver@^6.3.0, semver@^6.3.1:
+semver@6.3.0:
+  version "6.3.0"
+  resolved "https://registry.yarnpkg.com/semver/-/semver-6.3.0.tgz#ee0a64c8af5e8ceea67687b133761e1becbd1d3d"
+  integrity sha512-b39TBaTSfV6yBrapU89p5fKekE2m/NwnDocOVruQFS1/veMgdzuPcnOM34M6CwxW8jH/lxEa5rBoDeUwu5HHTw==
+
+semver@^6.0.0, semver@^6.1.1, semver@^6.1.2, semver@^6.3.0, semver@^6.3.1:
```
Comment (JamieB-gu) [inline yarn.lock:16193]: @jamesgorrie I'm not sure I follow this change 🤔. It looks like the net effect would be to downgrade the version of semver to 6.3.0, which was originally upgraded away from in #8715?

Comment (arelra) [reply yarn.lock:16193]: Good catch @JamieB-gu yes `@pm2/io@~5.0.0` is still using semver@6.3.0. In which case we need to keep the resolution.

Comment (arelra) [reply yarn.lock:16193]: Interestingly `pm2@5.3.0`'s dependency on `@pm2/io@~5.0.0` has been updated to address the vulnerability in `@pm2/io@5.0.2`:

https://github.com/keymetrics/pm2-io-apm/commit/c5996e3464b94fc95d3ff911160d51a599270403

As pm2 accepts a patch this should mean we can specifically install/upgrade `pm2/io@~5.0.2` to pick up this update?

Comment (jamesgorrie) [reply yarn.lock]: I think there has been a few issues where `yarn` was following `semver` so not updating it's lockfile as it was technically backwards compatible.

So basically the same thing you're talking about @arelra.

I removed `pm2` and re-added it which seems to have given us a more expected lockfile, but annoyingly we do still have high vulnerabilities, which I think are related to licensing.

**Review by JamieB-gu — APPROVED** [review]

**Comment by jamesgorrie** [comment]
As part of this we discovered we still have a `.snyk` file we should sort outhttps://github.com/guardian/dotcom-rendering/issues/8924

**Pull request merged**